rundeckapp/grails-app/conf/spring/resources.groovy

```diff
@@ -144,6 +149,23 @@ beans={
             log.warn("rundeck.multiURL enabled but no grails.serverURL found. This feature will be disabled.")
         }
     }
+
+    def defaultCookieSerializer = new DefaultCookieSerializer()
+
+    if (application.config.rundeck.security?.cookie?.domainNamePattern?.trim()) {
+        defaultCookieSerializer.setDomainNamePattern(application.config.rundeck.security.cookie.domainNamePattern.toString().trim());
+    }
+
+    def cookieName = application.config.rundeck.security?.cookie?.name?.trim() ?: "JSESSIONID"
+    defaultCookieSerializer.setCookieName(cookieName)
```

Comment:
the preferred way to define beans is with the bean builder syntax, instead of instantiating beans in this location

for example:

```.groovy
rundeckCookieSerializer(DefaultCookieSerializer){
    if (application.config.rundeck.security?.cookie?.domainNamePattern?.trim()) {
        domainNamePattern=application.config.rundeck.security.cookie.domainNamePattern.toString().trim()
    }
    cookieName = application.config.rundeck.security?.cookie?.name?.trim() ?: "JSESSIONID"
}
```

then for the session filter it would be something like this:

```.groovy
rundeckMapSessionRepository(MapSessionRepository, new ConcurrentHashMap())
rundeckSessionIdResolver(CookieHttpSessionIdResolver){
    cookieSerializer = ref('rundeckCookieSerializer')
}
springSessionRepositoryFilter(SessionRepositoryFilter, ref('rundeckMapSessionRepository')) {
        httpSessionIdResolver =  ref('rundeckSessionIdResolver')
}
```